login/views.py

```diff
@@ -0,0 +1,17 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.views import LoginView
+from django.http import HttpResponse
+from django.shortcuts import render
+
+
+# Create your views here.
+
+@login_required
+def example_dashboard(request):
+    # TOBEREMOVE this is for debugging
+    # return HttpResponse("It is example dashboard page.")
+    return render(request, 'login/dashboard.html')
+
+
+class EmloLoginView(LoginView):
+    template_name = 'login/login.html'
```

Comment:
this shows a very good understanding of Django @philipkcl, very well done

I haven't used class based views properly.

I think it would be better if you could write a function that handles and authorises the user and redirects him to the dashboard.

Does that make sense? Feel free to disagree if you think we should use your approach.

Reply:
> write a function that handles and authorises the user and redirects him to the dashboard.

No sure the meaning. `@login_required` should auto request `authorisation`

`Django` support two variable for `@login_required`, you can found those variables in `settings.py`
* with `@login_required`, if User does not login yet, User will be redirect to `LOGIN_URL`
* after User login, browser will be redirect back to previous page or `LOGIN_REDIRECT_URL`

Reply:
Good point Philip.

I was confused because that's not what happened when I tested the PR.

I could login and access the `@login_required` dashboard but was **NOT** redirected there automatically when logging in